features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Shared/CustomSplitButton.jsx

```diff
@@ -91,7 +92,7 @@ export default function CustomSplitButton(props) {
                                             <MenuItem
                                                 key={option}
                                                 selected={index === selectedIndex}
-                                                onClick={(event) => handleMenuItemClick(event, index)}
+                                                onClick={(event) => handleClick(event, index)}
```

Comment:
Using array index as a key to find an element is a bit flaky, Can't we use an id here, and also `key` should be a string. not an object, hope `option` is a string value.

> A “key” is a special string attribute you need to include when creating lists of elements.
[Docs](https://reactjs.org/docs/lists-and-keys.html#basic-list-component)

Reply:
Yes,
const options = ['Save and deploy', 'Save'];
Option is string value.